outputs/awssecuritylake.go

```diff
@@ -196,54 +198,80 @@ func getAWSSecurityLakeSeverity(priority types.PriorityType) (int32, string) {
 func (c *Client) EnqueueSecurityLake(falcopayload types.FalcoPayload) {
 	offset, err := c.Config.AWS.SecurityLake.Memlog.Write(c.Config.AWS.SecurityLake.Ctx, []byte(falcopayload.String()))
 	if err != nil {
-		if err.Error() != "future offset" {
-			go c.CountMetric(Outputs, 1, []string{"output:awssecuritylake.", "status:error"})
-			c.Stats.AWSSecurityLake.Add(Error, 1)
-			c.PromStats.Outputs.With(map[string]string{"destination": "awssecuritylake.", "status": Error}).Inc()
-			log.Printf("[ERROR] : %v SecurityLake. - %v\n", c.OutputType, err)
-			return
-		}
+		go c.CountMetric(Outputs, 1, []string{"output:awssecuritylake.", "status:error"})
+		c.Stats.AWSSecurityLake.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "awssecuritylake.", "status": Error}).Inc()
+		log.Printf("[ERROR] : %v SecurityLake. - %v\n", c.OutputType, err)
+		return
 	}
 	log.Printf("[INFO]  : %v SecurityLake. - Event queued (%v)\n", c.OutputType, falcopayload.UUID)
 	*c.Config.AWS.SecurityLake.WriteOffset = offset
 }
 
 func (c *Client) StartSecurityLakeWorker() {
+	awslake := c.Config.AWS.SecurityLake // assumes no concurrent r/w
+	ctx := awslake.Ctx
+	ml := awslake.Memlog
+
 	for {
-		time.Sleep(time.Duration(c.Config.AWS.SecurityLake.Interval) * time.Minute)
-		// time.Sleep(5 * time.Second)
-		batch := make([]memlog.Record, c.Config.AWS.SecurityLake.BatchSize)
-		count, err := c.Config.AWS.SecurityLake.Memlog.ReadBatch(c.Config.AWS.SecurityLake.Ctx, *c.Config.AWS.SecurityLake.ReadOffset+1, batch)
+		time.Sleep(time.Duration(awslake.Interval) * time.Minute)
+		batch := make([]memlog.Record, awslake.BatchSize)
+		count, err := ml.ReadBatch(ctx, *awslake.ReadOffset+1, batch)
 		if err != nil {
-			if err.Error() != "future offset" {
+			if errors.Is(err, memlog.ErrFutureOffset) {
+				// 	could be partial batch, continue batch logic below
+			}
+
+			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
 				go c.CountMetric(Outputs, 1, []string{"output:awssecuritylake.", "status:error"})
 				c.Stats.AWSSecurityLake.Add(Error, 1)
 				c.PromStats.Outputs.With(map[string]string{"destination": "awssecuritylake.", "status": Error}).Inc()
 				log.Printf("[ERROR] : %v SecurityLake. - %v\n", c.OutputType, err)
+				return
```

Comment:
Made this unrecoverable as I'm assuming ctx handling in main so cleaning up here?

Reply:
the worker will not be restarted, no recovery process for now, this is why we mus ignore and `continue`